Expose an option for rendering image/link URLs relative to a base

gibson042 opened the pull request.

joshbruce commented on test/tests/relative_urls.baseUrl=http%3A%2F%2Fexample%2Ecom%2Fbase%2F.text:
```diff
@@ -0,0 +1,22 @@
+# Absolutization of RFC 3986 URIs
```
Would it be possible to see a test covering dot-relative pathing?

```
./

../

../../../
```

For example; or does that go beyond bounds for this PR?

joshbruce commented:
ps. Hey. Thanks for contributing @gibson042! Do you normally contribute Marked? I'm looking to build a contributor team to establish a core so I don't become a bottle-neck on enhancements...we've been there already, yeah?

joshbruce commented:
@gibson042: Gonna have to do a pull I think to resolve the conflict.

gibson042 commented:
I rebased and added dot-relative tests. As for contributions, I have no significant prior involvement with the project, but would be happy to watch issues and pull requests for keeping down the bus factor.

joshbruce commented:
@gibson042: Right on. Thanks much! Hopefully, we can keep the quality up.
